hal/source/bmp180_device.cpp

```diff
@@ -0,0 +1,336 @@
+
+	return kNoError;
+}
+
+error_condition Bmp180Device::GetSystemParam(void) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	vector<uint8_t> buffer(kBUFFER_SIZE);
+	if (BusRead(kCHIP_ID_REG, kGEN_READ_WRITE_DATA_LENGTH, buffer) != kNoError)
+		return make_error_condition(errc::io_error);
+	chip_id_ = buffer[0];
+
+	oversamp_setting_ = kINITIALIZE_OVERSAMP_SETTING;
+	sw_oversamp_ = kINITIALIZE_SW_OVERSAMP;
+
+	if (BusRead(kVERSION_REG, kGEN_READ_WRITE_DATA_LENGTH, buffer) != kNoError)
+		return make_error_condition(errc::io_error);
+	ml_version_ = buffer[0] & 0x0F;
+	al_version_ = (buffer[0] & 0xF0) >> 4;
+
+	return kNoError;
+}
+
+error_condition Bmp180Device::GetCalibParam(void) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	vector<uint8_t> params_buffer(kCALIB_DATA_SIZE);
+	if (BusRead(kPROM_START_ADDR, kPROM_DATA_LEN, params_buffer) != kNoError)
+		return make_error_condition(errc::io_error);
+
+	calib_param_.ac1 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_AC1_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC1_LSB]);
+	calib_param_.ac2 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_AC2_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC2_LSB]);
+	calib_param_.ac3 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_AC3_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC3_LSB]);
+	calib_param_.ac4 = static_cast<unsigned int>(\
+			params_buffer[kCALIB_PARAM_AC4_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC4_LSB]);
+	calib_param_.ac5 = static_cast<unsigned int>(\
+			params_buffer[kCALIB_PARAM_AC5_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC5_LSB]);
+	calib_param_.ac6 = static_cast<unsigned int>(\
+			params_buffer[kCALIB_PARAM_AC6_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS| params_buffer[kCALIB_PARAM_AC6_LSB]);
+	calib_param_.b1 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_B1_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_B1_LSB]);
+	calib_param_.b2 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_B2_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_B2_LSB]);
+	calib_param_.mb = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_MB_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_MB_LSB]);
+	calib_param_.mc = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_MC_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_MC_LSB]);
+	calib_param_.md = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_MD_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_MD_LSB]);
+	return kNoError;
+}
+
+error_condition Bmp180Device::GetTemperature(int& temperature) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	int  uncomp_temperature, x1, x2 = 0;
```

Comment:
Please pay attention to white spaces, should be:
```suggestion
	int uncomp_temperature, x1, x2 = 0;
```